automerge/src/automerge.rs

```diff
@@ -27,76 +27,43 @@ pub struct Automerge {
     pub(crate) deps: HashSet<ChangeHash>,
     pub(crate) saved: Vec<ChangeHash>,
     pub(crate) ops: OpSet,
-    pub(crate) actor: Option<usize>,
+    pub(crate) actor: usize,
     pub(crate) max_op: u64,
 }
 
 impl Automerge {
     pub fn new() -> Self {
-        Automerge {
+        let mut am = Automerge {
             queue: vec![],
             history: vec![],
             history_index: HashMap::new(),
             states: HashMap::new(),
             ops: Default::default(),
             deps: Default::default(),
             saved: Default::default(),
-            actor: None,
+            actor: 0,
             max_op: 0,
-        }
+        };
+        am.set_random_actor();
+        am
     }
 
     pub fn set_actor(&mut self, actor: ActorId) {
-        self.actor = Some(self.ops.m.actors.cache(actor))
+        self.actor = self.ops.m.actors.cache(actor)
     }
 
-    fn random_actor(&mut self) -> ActorId {
+    fn set_random_actor(&mut self) {
```

Comment:
I wonder if this should actually be something like `with_random_actor(self) -> Self`. Would make the uses of it read a little nicer.

Reply:
This isn't there any more, resolving